lib/hiera/backend/eyaml/encryptors/gpg.rb

```diff
@@ -17,7 +17,7 @@ class Gpg < Encryptor
           self.options = {
             :gnupghome => { :desc => "Location of your GNUPGHOME directory",
                             :type => :string,
-                            :default => "#{ENV[["HOME", "HOMEPATH"].detect { |h| ENV[h] != nil }]}/.gnupg" },
+                            :default => ENV["HOME"] || ENV["HOMEPATH"] || ".gnupg" },
```

Comment:
If `$HOME` or `$HOMEPATH` is set then this means that `gnupghome` will be set explicitly to that environment variable instead of `$HOME/.gnupg` as expected.

Having the last stop default as `.gnupg` doesn't make sense to me - that would make it depend on your CWD.

I think the solution here is to simply throw a clear exception if neither `$HOME` nor `$HOMEPATH` are set and the directory isn't set explicitly.